src/math/p5.Vector.js

```diff
@@ -1440,6 +1440,31 @@ p5.Vector.prototype.heading = function heading() {
   return h;
 };
 
+/**
+ * Rotate the vector to a specific angle (only 2D vectors), magnitude remains the
+ * same
+ *
+ * @method setHeading
+ * @param  {number}    angle the angle of rotation
+ * @chainable
+ * @example
+ * <div class="norender">
+ * <code>
+ * let v = createVector(10.0, 20.0);
+ * // result of v.heading() is 1.1071487177940904
+ * v.setHeading(Math.PI);
+ * // result of v.heading() is now 3.141592653589793
+ * </code>
+ * </div>
+ */
+
+p5.Vector.prototype.setHeading = function setHeading(a) {
```

Comment:
to address @montoyamoraga's point, let's add a friendly error message check into this one so if people try to use it with 3D vectors it will give an error message to help debug. you can see an example of how to do this [here](https://github.com/processing/p5.js/blob/main/src/core/shape/2d_primitives.js#L173)

Reply:
I'm a little confused how the FES system would work here. How would one use the FES system to check if the 'z' component of the object is undefined?

Reply:
you just add the line:
```js
p5._validateParameters('p5.Vector.setHeading', arguments);
```
and it takes care of itself, checking against the @param documentation for this method that you've added

Reply:
ah sorry scratch that, i thought the vector was passed as an argument. so this won't work.

Reply:
No worries. I'm going to push the unit tests and hopefully we can get the FES up and running